src/credential/presentation/errors.ts

```diff
@@ -0,0 +1,41 @@
+import { IoWalletError, serializeAttrs } from "../../utils/errors";
+
+/**
+ * An error subclass thrown when auth request decode fail
+ *
+ */
+export class AuthRequestDecodeError extends IoWalletError {
+  code = "ERR_IO_WALLET_AUTHENTICATION_REQUEST_DECODE_FAILED";
+
+  /** The Claim for which the validation failed. */
+  claim: string;
+
+  /** Reason code for the validation failure. */
+  reason: string;
+
+  constructor(
+    message: string,
+    claim: string = "unspecified",
+    reason: string = "unspecified"
+  ) {
+    super(serializeAttrs({ message, claim, reason }));
+    this.claim = claim;
+    this.reason = reason;
+  }
+}
```

Comment:
what do you think about?
```suggestion
export class AuthRequestDecodeError extends IoWalletError {
  static readonly CODE = "ERR_IO_WALLET_AUTHENTICATION_REQUEST_DECODE_FAILED";

  readonly claim: string;
  readonly reason: string;

  constructor(message: string, claim = "unspecified", reason = "unspecified") {
    super(serializeAttrs({ message, claim, reason }));
    this.claim = claim;
    this.reason = reason;
  }

  get code() {
    return AuthRequestDecodeError.CODE;
  }
}
```

Reply:
This error was moved from another folder; the only modification was the removal of the static property. It's also in the presentation section, which is outdated, not used and will probably require future changes. If that's ok for you, I would leave the file as is.